srcpkgs/dpdk/template

```diff
@@ -0,0 +1,87 @@
+# Template file for 'dpdk'
+pkgname=dpdk
+version=19.08
+revision=1
+archs="x86_64 i686"
```

Comment:
Judging by the documentation, this should also work on other architectures. So it is preferred to use `broken=` or `nocross=` with the reason why it doesn't work for us.

Reply:
I will try again to compiler on the other platforms. Didn't do well on the first attempts.

Reply:
These are the errors I'm getting on the different platforms.

aarch64, armv7, armv6l-musll: Exec format error: '/builddir/dpdk-19.08/build/meson-private/sanitycheckc.exe'.
x86_64-musl, : execinfo.h: No such file or directory #include <execinfo.h>

The problem with the exec format error seems to be a meson bug. It compiles for the wrong target platform. I'm not sure if we are cross-compiling in Travis.

Reply:
> The problem with the exec format error seems to be a meson bug. It compiles for the wrong target platform. I'm not sure if we are cross-compiling in Travis.

meson build_style handles cross compiling. I checked locally, and it compiled fine with it after adding `-Db_lto=false` (only checked on `x86_64` and cross compiling to `aarch64`)
```patch
diff --git a/srcpkgs/dpdk/template b/srcpkgs/dpdk/template
index a0eff47adf..bb1c37cf95 100644
--- a/srcpkgs/dpdk/template
+++ b/srcpkgs/dpdk/template
@@ -2,9 +2,9 @@
 pkgname=dpdk
 version=19.08
 revision=1
-archs="x86_64 i686"
-build_style=meta
-hostmakedepends="cmake meson doxygen python3-Sphinx pkg-config elfutils"
+build_style=meson
+configure_args="-Denable_kmods=false -Denable_docs=true -Db_lto=false"
+hostmakedepends="doxygen python3-Sphinx pkg-config elfutils"
 makedepends="libnuma-devel linux-headers jansson-devel libpcap-devel zlib-devel libressl-devel libbsd-devel libelf libmnl"
 depends="jansson zlib libbsd libressl python-pyelftools"
 short_desc="Data Plane Development Kit"
@@ -14,24 +14,20 @@ homepage="https://www.dpdk.org/"
 distfiles="https://github.com/DPDK/${pkgname}/archive/v${version}.tar.gz"
 checksum=1ceff1a6f4f8d5f6f62c1682097249227ac5225ccd9638e0af09f5411c681038
 
-do_configure() {
-	meson --prefix /usr -Denable_kmods=false -Denable_docs=true build
-}
-
-do_build() {
-	ninja -C build
-}
-
-do_install() {
-	DESTDIR=${DESTDIR} ninja -C build install
-	vmkdir usr/lib/modules-load.d
-}
+if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+	broken="Usage of error.h"
+fi
 
 post_install() {
+	vmkdir usr/lib/modules-load.d
 	vlicense license/bsd-3-clause.txt
 	vlicense license/gpl-2.0.txt
 }
 
+do_check() {
+	:
+}
+
 dpdk-doc_package() {
 	short_desc+=" - documentation"
 	archs=noarch
```

> x86_64-musl, : execinfo.h: No such file or directory #include <execinfo.h>

this could be fixed by adding libexecinfo-devel to makedepends. But in my local test it would error out later due to usage of `error.h`. So add 
```
if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
        broken="Usage of error.h"
fi
```

Reply:
Should it say: `broken="Usage of execinfo.h"` ?

Reply:
Architecture armv7l throws error:  `config/arm/meson.build:196:7: ERROR: Could not get define '__ARM_NEON'`

No idea what it could mean.

Reply:
> Architecture armv7l throws error: `config/arm/meson.build:196:7: ERROR: Could not get define '__ARM_NEON'`
> 
> No idea what it could mean.

This normally should just not set `RTE_CPUFLAG_NEON` instead of erroring out. Set it to broken for now. I can take a look at it later. The meson docs does not document that it should throw any erros
> get_define(definename) returns the given preprocessor symbol's value as a string or empty string if it is not defined.